### PR TITLE
[v2] Make all of emission use Relay

### PR DESCRIPTION
**Pull request by alloy**
Relies on artsy/metaphysics#1922 & https://github.com/artsy/metaphysics/pull/1924

Closes https://artsyproduct.atlassian.net/browse/PLATFORM-1385
Fixes https://artsyproduct.atlassian.net/browse/PLATFORM-1735
Part of https://artsyproduct.atlassian.net/browse/PLATFORM-1734

#skip_new_tests

### CODE REVIEW

File: src/lib/Components/Bidding/Context/TimeOffsetProvider.tsx [hunk 1]
```diff
@@ -1,41 +1,35 @@
...
+  )
 
 const getGravityTimestampInMilliSeconds = async () => {
   const startTime = getLocalTimestampInMilliSeconds()
-  const { data } = await fetchSystemTime()
+  const { system } = await fetchSystemTime()
 
   const possibleNetworkLatencyInMilliSeconds = (getLocalTimestampInMilliSeconds() - startTime) / 2
-  const serverTimestampInMilliSeconds = data.system.time.unix * 1e3 + possibleNetworkLatencyInMilliSeconds
+  const serverTimestampInMilliSeconds = system.time.unix * 1e3 + possibleNetworkLatencyInMilliSeconds
```
Comment (alloy) [inline src/lib/Components/Bidding/Context/TimeOffsetProvider.tsx]: cc @yuki24 @dleve123 Something to keep an eye on during next QA.

File: src/lib/Components/Consignments/Screens/ConsignmentsArtist.tsx [hunk 2]
```diff
@@ -41,31 +42,38 @@ export default class Artist extends React.Component<Props, State> {
...
+    const data = await fetchQuery<ConsignmentsArtistQuery>(
       environment,
       graphql`
         query ConsignmentsArtistQuery($query: String!) {
-          matchArtist(term: $query) {
-            internalID
-            name
-            image {
-              url
+          searchConnection(query: $query, first: 10, entities: [ARTIST], mode: AUTOSUGGEST) {
```
Comment (mzikherman) [inline src/lib/Components/Consignments/Screens/ConsignmentsArtist.tsx]: Sneaky! This is a good a time as any to clarify the difference here.

@cavvia , we have 2 ways of getting what seems to be the same/v similar data. One is to use `/match/suggest?indexes[]=Artist` (global auto suggest restricted to just artists), and the other is to use `/match/artists`.

It's hard for me to understand the difference between the two, the only clear explicit difference seems to be that the auto-suggest one searches alternate names as well. But there might be other functional differences lurking beneath the hood.

File: src/lib/Components/Consignments/Submission/Gemini/createGeminiAssetWithS3Credentials.ts [hunk 3]
```diff
@@ -0,0 +1,36 @@
+import {
+  createGeminiAssetWithS3CredentialsMutation,
+  CreateGeminiEntryForAssetInput,
+} from "__generated__/createGeminiAssetWithS3CredentialsMutation.graphql"
+import { defaultEnvironment } from "lib/relay/createEnvironment"
+import { commitMutation, graphql } from "relay-runtime"
+
+export const createGeminiAssetWithS3Credentials = (input: CreateGeminiEntryForAssetInput) => {
```
Comment (mzikherman) [inline src/lib/Components/Consignments/Submission/Gemini/createGeminiAssetWithS3Credentials.ts]: I like how you can use Relay (and tooling) and not need to be in the context of an actual Relay container. ie- you can use it just in helpers/utils too

File: src/lib/Components/Consignments/Submission/addAssetToConsignment.ts [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+            asset {
+              submissionID
+            }
+          }
+        }
+      `,
+      variables: {
+        input: {
+          ...input,
+          clientMutationId: Math.random().toString(8),
```
Comment (mzikherman) [inline src/lib/Components/Consignments/Submission/addAssetToConsignment.ts]: Is this needed?

Comment (alloy) [reply src/lib/Components/Consignments/Submission/addAssetToConsignment.ts]: I actually don’t believe so, I just wanted to port the existing code with as few changes a possible. I’m open to revisiting, though.

File: src/lib/Components/Consignments/Screens/ConsignmentsArtist.tsx (hunk 2 above)
Comment (cavvia) [reply src/lib/Components/Consignments/Screens/ConsignmentsArtist.tsx]: I would expect suggest to use the auto-suggest query so it's designed for completion of an active query (which is being typed out) rather than a finalised query term.

File: src/lib/Components/Bidding/Context/TimeOffsetProvider.tsx (hunk 1 above)
Comment (dleve123) [reply src/lib/Components/Bidding/Context/TimeOffsetProvider.tsx]: Can you provide some more context here? Can you summarize the relevant diff, and provide perspective on any risk here?

Comment (alloy) [reply src/lib/Components/Bidding/Context/TimeOffsetProvider.tsx]: I changed this system time provider, used in bidding views, to use a different fetch mechanism. There should be _no_ functional differences, but still a good thing to go through the bidding views next QA (as you would normally do anyways).

File: src/lib/Components/Consignments/Submission/addAssetToConsignment.ts (hunk 4 above)
Comment (alloy) [reply src/lib/Components/Consignments/Submission/addAssetToConsignment.ts]: Let's leave this as-is for now, next time when I come across it and we don't already have so many changes in flight I'll take a stab.

**Comment by alloy** [comment]
@mzikherman Sounds like we’re good to go wrt the artist search, right?

File: src/lib/Components/Consignments/Screens/ConsignmentsArtist.tsx (hunk 2 above)
Comment (mzikherman) [reply src/lib/Components/Consignments/Screens/ConsignmentsArtist.tsx]: Interesting! I feel like all the use cases I've seen of `/match/artists` have generally been for auto-completes. But we can revisit if it comes up again!

File: src/lib/Components/Consignments/Submission/addAssetToConsignment.ts (hunk 4 above)
Comment (mzikherman) [reply src/lib/Components/Consignments/Submission/addAssetToConsignment.ts]: Totally makes sense 👍

**Comment by mzikherman** [comment]
Let's go for it.
